tests/Application/StripeFormattingTrait.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace MatchBot\Tests\Application;
+
+use ArrayIterator;
+use Prophecy\PhpUnit\ProphecyTrait;
+use Prophecy\Prophecy\ObjectProphecy;
+use Stripe\Collection;
+use Stripe\StripeObject;
+
+trait StripeFormattingTrait
+{
+    use ProphecyTrait;
+
+    protected function buildAutoIterableCollection(string $json): Collection|ObjectProphecy
+    {
+        /** @var \stdClass $itemsArray */
+        $itemsArray = json_decode($json, false);
+        /** @var StripeObject[] $itemData */
+        $itemData = $itemsArray->data;
+
+        $iterableCollection = $this->prophesize(Collection::class);
+        $iterableCollection->autoPagingIterator()->willReturn(new ArrayIterator($itemData));
+        $iterableCollection->count()->willReturn(count($itemData));
+
+        return $iterableCollection->reveal();
+    }
```

Comment:
Slight simplifications:

```suggestion
    private function buildAutoIterableCollection(string $json): Collection
    {
        $itemData = json_decode($json, false, 512, JSON_THROW_ON_ERROR)->data;

        $iterableCollection = $this->prophesize(Collection::class);
        $iterableCollection->autoPagingIterator()->willReturn($itemData);
        $iterableCollection->count()->willReturn(count($itemData));

        return $iterableCollection->reveal();
    }
```

- Private functions in a trait can still be accessed by classes that use the trait
- Returning the ArrayIterator doesn't seem to add anything over just iterating the array directly
- This function always returns a Collection, never an ObjectProphecy.

Similar applies to functions below.